homeassistant/bootstrap.py

```diff
@@ -380,38 +385,44 @@ def _get_domains(hass: core.HomeAssistant, config: dict[str, Any]) -> set[str]:
     return domains
 
 
-async def _async_log_pending_setups(
-    hass: core.HomeAssistant, domains: set[str], setup_started: dict[str, datetime]
-) -> None:
+async def _async_watch_pending_setups(hass: core.HomeAssistant) -> None:
     """Periodic log of setups that are pending for longer than LOG_SLOW_STARTUP_INTERVAL."""
+    loop_count = 0
+    setup_started: dict[str, datetime] = hass.data[DATA_SETUP_STARTED]
     while True:
```

Comment:
Wonder if we can change this to `while not hass.is_running:` so we don't have to cancel it.

Reply:
`setup_started` can be empty between phases so its possible if it hits just right it would terminate prematurely.